examples/tcp_ca/build.rs

```diff
@@ -8,9 +8,12 @@ use libbpf_cargo::SkeletonBuilder;
 const SRC: &str = "src/bpf/tcp_ca.bpf.c";
 
 fn main() {
-    let mut out =
-        PathBuf::from(env::var_os("OUT_DIR").expect("OUT_DIR must be set in build script"));
-    out.push("tcp_ca.skel.rs");
+    let out = PathBuf::from(
+        env::var_os("CARGO_MANIFEST_DIR").expect("CARGO_MANIFEST_DIR must be set in build script"),
+    )
+    .join("src")
+    .join("bpf")
+    .join("tcp_ca.skel.rs");
```

Comment:
No and no: I want to see what is generated. I don't care if `git status` reports untracked files. In fact, that's what I expect it to.

Reply:
>  I don't care if git status reports untracked files.

I'd expect auto-generated files to be part of .gitignore, so that no one (doesn't matter if it's you or any other contributor) could accidentally include them. That's a typical set up in any other repository, why a sudden deviation here?

Reply:
From my perspective the skeleton is basically one of the APIs that users program against. Without seeing the skeleton, it's non-obvious how to interface with various types. However, it's equally non-obvious that such a thing even exists. Having it show up *somewhere* is key here. You mention end user focus several times. I argue that discoverability is crucial to that. Not ignoring the files would help with that in a way that at least I find to be the most appealing.

There is another wrinkle to it, which is what happens when we use the `cargo libbpf` sub-command (e.g., `cargo run -- libbpf make --clang-args='-I../examples/vmlinux/x86_64/'`): it will generate skeletons next to the `.bpf.c` files (as this change does more programmatically), but there the expectation absolutely is that such newly created files are visible. If we ignore them with this change, we will make them invisible there as well.

I don't intend to die on this hill, and I can certainly see arguments in favor of either. That said, given that these skeletons only show up if a user builds a specific example, I think not ignoring them is fine and makes for a better experience.

Reply:
We discussed this a bit offline. It seems like `cargo libbpf` already documents skeleton generation (in `## gen` section) to some degree, though documentation can always be improved.

BTW, would skeleton be generated without explicit SkeletonBuilder invocations in build.rs? Because skeleton is not really mandatory  (.bpf.o generation is the only required step), so I'm just wondering what's happening if user doesn't write a custom build.rs? If user has to call into SkeletonBuilder explicitly, then I think there is no issue of discoverability, even with example user would have to check .bpf.c, .rs, and build.rs code to put all the things together, no?

Anyways, this is not something that has to be set in stone, and we can revisit it later, so up to you what to do here. It's just unconventional to have auto-generated files that are not listed in .gitignore, effectively looking like forgotten left over files.

Reply:
Skeleton files are now in `gitignore` as of https://github.com/libbpf/libbpf-rs/commit/545c59c006ebb9ade251cbf34ea4e249b12c9f2f.